### PR TITLE
ENH - Add fixed-point distance strategy to build working sets in ``ProxNewton``

### LINKED ISSUE #100: ENH - Add fixpoint strategy for Prox-Newton solver
Following https://github.com/a-rahimi/sparse-regression/blob/main/sparse%20regression.ipynb, the current prox-Newton solver does not handle fixpoint strategy, hence we cannot use this solver to solve l05 or l23 penalized problems

### PR DESCRIPTION
**Pull request by PABannier**
## Context of the PR

``ProxNewton`` does not support a fixed-point strategy for building the working set. This limitation makes some problems out-of-reach, for instance, Sparse Group Poisson regression.

Closes #100

## Contributions of the PR

- add support of ``ws_strategy="fixpoint"`` for ``ProxNewton`` solver.

### Checks before merging PR

- [x] added documentation for any new feature
- [x] added unittests
- [x] edited the [what's new](../doc/changes/whats_new.rst)

### CODE REVIEW

File: skglm/solvers/prox_newton.py [hunk 1]
```diff
@@ -86,7 +92,11 @@ def solve(self, X, y, datafit, penalty, w_init=None, Xw_init=None):
             else:
                 grad = _construct_grad(X, y, w[:n_features], Xw, datafit, all_features)
 
-            opt = penalty.subdiff_distance(w[:n_features], grad, all_features)
+            if self.ws_strategy == "subdiff":
```
Comment (Badr-MOUFAD) [inline skglm/solvers/prox_newton.py:95]: ``fixpoint`` should be also considered when determining the descent direction

https://github.com/scikit-learn-contrib/skglm/blob/4cf5976b11563e49e76c4b880521bdd00e2e8fa8/skglm/solvers/prox_newton.py#L218

**Review by Badr-MOUFAD — COMMENTED** [review]
I don't know whether it is still WIP but here are some remarks.

Also, could you add ``fixpoint`` to the `ProxNewton` unittests?

**Comment by mathurinm** [comment]
Looks nice @PABannier, this will definitely improve UX!

From an API point of view, shouldn't this check be delegated to each solver? This way we don't have one big function, but `Solver.validate(datafit, penalty)`, in the spirit of what @Badr-MOUFAD implemented here : https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/experimental/pdcd_ws.py#L201

Such functions could also take care of the initialization (e.g. stepsize computation)  which is done on a solver basis. WDYT?

**Review by Badr-MOUFAD — APPROVED** [review]
LGTM :rocket:

Many thanks @PABannier for this great job!
Your feedback on the last changes is very valuable before merging

**Comment by PABannier** [comment]
@Badr-MOUFAD LGTM as well! Thanks for finishing it :)

**Pull request merged**